homeassistant/components/recorder/statistics.py

```diff
@@ -858,14 +859,19 @@ def list_statistic_ids(
             hass, statistic_ids=statistic_ids, statistic_type=statistic_type
         )
 
-        for statistic_id, info in platform_statistic_ids.items():
-            if (unit := info["unit_of_measurement"]) is not None:
-                # Display unit according to user settings
-                unit = _configured_unit(unit, units)
-            platform_statistic_ids[statistic_id]["unit_of_measurement"] = unit
-
-        for key, value in platform_statistic_ids.items():
-            result.setdefault(key, value)
+        for key, meta in platform_statistic_ids.items():
+            if key in result:
+                continue
+            result[key] = {
+                "has_mean": meta["has_mean"],
+                "has_sum": meta["has_sum"],
+                "name": meta["name"],
+                "source": meta["source"],
+                "display_unit_of_measurement": _display_unit(
+                    hass, meta["unit_of_measurement"]
+                ),
+                "unit_of_measurement": meta["unit_of_measurement"],
```

Comment:
Should we flip this? To better match the entity models?

So, 

- `unit_of_measurement` -> `native_unit_of_measurement`
- `display_unit_of_measurement` -> `unit_of_measurement`

?

Reply:
The units used here are not the same as in the entity model where we have a unit native to the sensor (entity property `native_unit_of_measurement`) and a unit used in the state (state attribute `unit_of_measurement`), instead we have:
- The unit used when storing statistics - named `statistics_unit_of_measurement` in this PR
- The unit used when querying the statistics -